src/test/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetailsTest.java

```diff
@@ -225,4 +226,95 @@ public void testValidateResponsesDetails_noValidationError_errorListShouldBeEmpt
 
         assertEquals(0, errors.size());
     }
+
+    @Test
+    public void testIsQuestionDropdownEnabled_shouldReturnTrue() {
+        FeedbackMcqQuestionDetails feedbackQuestionDetails = new FeedbackMcqQuestionDetails();
+        feedbackQuestionDetails.setQuestionDropdownEnabled(true);
+        assertTrue(feedbackQuestionDetails.isQuestionDropdownEnabled());
+    }
+
+    @Test
+    public void testSetQuestionDropdownEnabled_shouldReturnFalse() {
+        FeedbackMcqQuestionDetails feedbackQuestionDetails = new FeedbackMcqQuestionDetails();
+        feedbackQuestionDetails.setQuestionDropdownEnabled(false);
+        assertFalse(feedbackQuestionDetails.isQuestionDropdownEnabled());
+    }
+
+    @Test
+    public void testValidateQuestionDetails_weightsNotEnabledButWeightListNotEmpty_errorReturned() {
+        FeedbackMcqQuestionDetails mcqDetails = new FeedbackMcqQuestionDetails();
+        mcqDetails.setMcqWeights(Arrays.asList(1.22, -1.55));
+        mcqDetails.setMcqChoices(Arrays.asList("choice 1", "choice 2"));
+        mcqDetails.setHasAssignedWeights(false);
+        List<String> errors = mcqDetails.validateQuestionDetails();
+        assertEquals(FeedbackParticipantType.NONE, mcqDetails.getGenerateOptionsFor());
```

Comment:
I think you can move this assert to the test which is testing the constructor, and maybe create another test to check the setter of this field.